src/aznfswatchdog

```diff
@@ -307,13 +307,49 @@ run_daily_once()
     fi
 }
 
+#
+# We only use lowercase single word names for distro id:
+# ubuntu, centos, redhat, sles.
+#
+canonicalize_distro_id()
+{
+    local distro_lower=$(echo "$1" | tr '[:upper:]' '[:lower:]')
+
+    # Use sles for SUSE/SLES.
+    if [ "$distro_lower" == "suse" ]; then
+        distro_lower="sles"
+    fi
+
+    echo "$distro_lower"
+}
+
+detect_aznfs_version()
+{
+    local distro="$distro_id"
+
+    sleep 45
+
+    if [ "$distro" == "ubuntu" ]; then
+        current_version=$(apt-cache show aznfs 2>/dev/null | grep "^Version" | tr -d " " | cut -d ':' -f2)
+    elif [ "$distro" == "centos" -o "$distro" == "rocky" -o "$distro" == "rhel" ]; then
+        current_version=$(yum info aznfs 2>/dev/null | grep "^Version" | tr -d " " | cut -d ':' -f2)
+    elif [ "$distro" == "sles" ]; then
+        current_version=$(zypper info aznfs_sles 2>/dev/null | grep "^Version" | tr -d " " | cut -d ':' -f2 | cut -d '-' -f1)
+    fi
+
+    vecho "AZNFS version: $current_version"
+}
+
 vecho "Starting aznfswatchdog..."
 
-# Detect and log distro and bash version
+# Detect and log distro, bash and AZNFS-mount version
```

Comment:
move L346 to 361 into a single function and call it log_version_info()
don't need the detect_aznfs_version() function and also don't do it in the background.

Reply:
background as it takes few seconds before we are able to fetch the aznfs version after starting watchdog, specifically for sles.